CONTRIBUTING.md

```diff
@@ -474,9 +476,59 @@ To add an example:
      };
    };
    ```
+
 4. Ensure the example works by building and running its test.
 <!--TODO: Reference the tests section when it is written -->
 
+## How to add a project demo
+
+A demo is a practical demonstration of an application, providing an easy way for users to test its functionality and assess its suitability for their use cases.
+
+1. To start, implement a NixOS module that adheres to the [demo type](./projects/types.nix):
+
+   ```nix
+   nixos.demo.TYPE = {
+     module = ./path/to/application/configuration.nix;
+     demo-stuff = ./path/to/demo/only/configuration.nix;
+     description = ''
+       Instructions for using the application
+
+       1.
+       2.
+       3.
+     '';
+     tests = { };
+   };
+   ```
+
+   - `TYPE` can be either `vm` or `shell`, indicating the preferred environment for running the application: a NixOS VM or a terminal shell.
+   - `module` is meant for setting up the application, while `demo-stuff` is for demo-specific things, like [demo-shell](./overview/demo/shell.nix) configuration.
```

Comment:
We sure need to rename that temporary `demo-stuff` XD

Reply:
Indeed, I kept it like that just to be consistent until we agree on something else.